src/metpy/calc/thermo.py

```diff
@@ -1023,16 +1023,36 @@ def saturation_vapor_pressure(temperature):
     .. math:: 6.112 e^\frac{17.67T}{T + 243.5}
 
     """
-    # Converted from original in terms of C to use kelvin.
-    return mpconsts.nounit.sat_pressure_0c * np.exp(
-        17.67 * (temperature - 273.15) / (temperature - 29.65)
-    )
+
+    def liquid(temperature):
+        # Converted from original in terms of C to use kelvin.
+        return mpconsts.nounit.sat_pressure_0c * np.exp(
+            17.67 * (temperature - 273.15) / (temperature - 29.65)
+        )
```

Comment:
Given this is what was there before, why not rename `saturation_vapor_pressure` to `_saturation_vapor_pressure_over_liquid`, create a new `_saturation_vapor_pressure_over_ice`, and create a new `saturation_vapor_pressure` that just dispatches to one of those?

It would allow skipping the dispatch if something needed one or the other

Reply:
Found [request by maintainer ](https://github.com/Unidata/MetPy/issues/2320#issuecomment-1026155013) to avoid new functions.

Perhaps replace the function definition line with `if phase != "ice"`, assign the result to `saturation_over_liquid`, and return that immediately if `phase == "liquid"`?  (and similarly for ice just below)